### ISSUE TITLE
[BUG]Failed to call forward() multiple times using deepspeed zero3 

### ISSUE THREAD

**Issue by Mark-divinci** [microsoft/DeepSpeed#5321]
**Describe the bug**
I want to call forward() multiple times to calculate different loss using deepspeed zero3 ,but I get this error.
"assert len(set(p.ds_id for p in self.params_in_ipg_bucket)) == len(self.params_in_ipg_bucket)
AssertionError"
It seems like caused by call model twice in forward,but l don't know how to fix it. 

**Expected behavior**
 ```
 def run_model():
      ......
      loss1 = self.model(input1)
      loss2 = self.model(input2)
      loss = loss1 + loss2
      return {"loss":loss}
```
  ```
def run():
      out = self.run_model(batch)
      loss = out["loss"]
      self.model.backward(loss)
      self.model.step()
```

**Screenshots**
<img width="1436" alt="image" src="https://github.com/microsoft/DeepSpeed/assets/140576398/bb94d68f-fdfa-481b-afed-433d5325bfc7">

**System info (please complete the following information):**
 deepspeed version 0.11.1

**Comment by Minami-su** [comment]
+1
